probe_builder/__init__.py

```diff
@@ -162,7 +162,18 @@ def build(builder_image_prefix,
 
     crawler_filter = kernel_crawler.repo.CrawlerFilter(machine=machine, arch=arch, distro_filter=distro_filter, kernel_filter=kernel_filter)
 
-    kernels = distro_obj.get_kernels(workspace, package, download_config, crawler_filter)
+    cache_file = "cache.yaml"
+    cache = kernelcache.KernelCache(cache_file)
+    cache_key = kernelcache.KernelCache.make_key(kernel_type, machine, arch, distro_filter, kernel_filter)
+    print("Looking up {} in cache".format(cache_key))
+
+    if cache.get(cache_key):
+        kernels = cache.get(cache_key)
```

Comment:
I will be hated by almost half the world.
```suggestion
    if kernels := cache.get(cache_key):
        pass
```